src/ds/thread_ids.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include <map>
+#include <thread>
+
+namespace threading
+{
+  extern std::map<std::thread::id, uint16_t> thread_ids;
+
+  static inline uint16_t get_current_thread_id()
+  {
+    if (thread_ids.empty())
+    {
+      return 0;
+    }
+
+    const auto tid = std::this_thread::get_id();
+    const auto it = thread_ids.find(tid);
+    if (it == thread_ids.end())
+    {
+      throw std::runtime_error(
```

Comment:
Any value in printing the thread ID here?

Reply:
I think very little - we're either fully initialised or we're not, and we deliberately never use the raw `thread_id` so we have nothing to associate this with.

Reply:
On the other hand, more information can't be a _bad_ thing, so I'll add it.